frontend/src/pages/destructionlist/review/DestructionListReview.action.ts

```diff
@@ -25,43 +66,145 @@ export const destructionListReviewAction = async ({
+  await addToZaakSelection(
+    getDestructionListReviewKey(destructionList),
+    [zaak],
+    { approved: true },
+  );
+  return null;
+}
 
-  const [zaakSelection] = (await Promise.all(promises)) as [ZaakSelection];
+/**
+ * React Router action, user intends to exclude an item.
+ * @param request
+ * @param params
+ */
+export async function destructionListExcludeItemAction({
+  request,
+}: ActionFunctionArgs) {
+  const { payload } = await request.json();
+  const { comment, destructionList, zaak } =
+    payload as ReviewDestructionListZaakActionPayLoad;
+  await addToZaakSelection(
+    getDestructionListReviewKey(destructionList),
+    [zaak],
+    { approved: false, comment },
+  );
+  return null;
+}
 
-  const zaakSelectionValid = Object.values(zaakSelection).filter(
-    (f) => f.selected,
+/**
+ * React Router action, user intends to unselect a selection of items.
+ * @param request
+ * @param params
+ */
+export async function destructionListUnselectItemsAction({
+  request,
+}: ActionFunctionArgs) {
+  const { payload } = await request.json();
+  const { destructionList, zaken } =
+    payload as ReviewDestructionListZakenActionPayLoad;
+  await removeFromZaakSelection(
+    getDestructionListReviewKey(destructionList),
+    zaken,
   );
+  return null;
+}
+
+/**
+ * React Router action, user intends to reject a list.
+ * @param request
+ * @param params
+ */
+export async function destructionListApproveListAction({
+  request,
+}: ActionFunctionArgs) {
+  const { payload } = await request.json();
+  const { destructionList } =
+    payload as ReviewDestructionListListApproveActionPayLoad;
 
   const data: Review = {
-    destructionList: destructionListUuid,
-    decision: zaakSelectionValid.length > 0 ? "rejected" : "accepted",
-    listFeedback: details.listFeedback,
-    zakenReviews: zaakSelectionValid.map((zaak) => {
-      if (!zaak.detail) {
-        throw new Error("Details are missing for one or more zaken");
-      }
-      const detail = zaak.detail as FormDataState;
-
-      return {
-        zaakUrl: detail.url,
-        feedback: detail.motivation,
-      };
-    }),
+    destructionList: destructionList,
+    decision: "accepted",
+    listFeedback: "",
   };
+
   await createDestructionListReview(data);
+  await clearZaakSelection(getDestructionListReviewKey(destructionList));
   return redirect("/");
-};
+}
+
+/**
+ * React Router action, user intends to reject a list.
+ * @param request
+ * @param params
+ */
+export async function destructionListRejectListAction({
+  request,
+}: ActionFunctionArgs) {
+  const { payload } = await request.json();
+  const { comment, destructionList, zaakReviews } =
+    payload as ReviewDestructionListListRejectActionPayLoad;
+
+  const data: Review = {
+    destructionList: destructionList,
+    decision: "rejected",
+    listFeedback: comment,
+    zakenReviews: zaakReviews,
+  };
+
+  await createDestructionListReview(data);
```

Comment:
Is there a possibility in making these in parallel? Or is there a specific reason they're sequential?